x-pack/plugins/fleet/public/applications/integrations/sections/epm/screens/detail/policies/actions_menu.tsx

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { memo, useState, useMemo } from 'react';
+import { EuiPortal, EuiContextMenuItem } from '@elastic/eui';
+import { FormattedMessage } from '@kbn/i18n/react';
+
+import type { AgentPolicy, PackagePolicy } from '../../../../../types';
+import {
+  AgentEnrollmentFlyout,
+  ContextMenuActions,
+  DangerEuiContextMenuItem,
+  PackagePolicyDeleteProvider,
+} from '../../../../../components';
+import { useAgentPolicyRefresh, useLink } from '../../../../../hooks';
+
+export const PackagePoliciesActionMenu: React.FunctionComponent<{
+  agentPolicy: AgentPolicy;
+  packagePolicy: PackagePolicy;
+}> = memo(({ agentPolicy, packagePolicy }) => {
+  const [isEnrollmentFlyoutOpen, setIsEnrollmentFlyoutOpen] = useState(false);
+  const { getHref } = useLink();
+
+  const onClose = useMemo(() => {
+    return () => setIsEnrollmentFlyoutOpen(false);
+  }, []);
+
+  const refreshAgentPolicy = useAgentPolicyRefresh();
+
+  return (
+    <>
+      {isEnrollmentFlyoutOpen && (
+        <EuiPortal>
+          <AgentEnrollmentFlyout agentPolicies={[agentPolicy]} onClose={onClose} />
+        </EuiPortal>
+      )}
+
+      <ContextMenuActions
```

Comment:
It's not super clear in the designs, but I think this context menu should be exactly the same as the one used in Fleet > Policies: https://github.com/elastic/kibana/blob/0894f67df90c4cb30913af81f370a57394214132/x-pack/plugins/fleet/public/applications/fleet/sections/agent_policy/details_page/components/package_policies/package_policies_table.tsx#L171

That means the new "Add agent" option should appear there too, and the "Delete integration" here should have the `is_managed` conditional. It would probably be best to share code between the two places.

Reply:
+1 that's the idea. One thing to note is that the "Add Agent" action will link to a flyout that is slightly different than the one users see in Fleet. For Integrations UI, we remove a step for selecting an agent policy, and the last step will point users to view integration assets / point to a specific kibana plugin. How we implement that though is up to you. More info here https://github.com/elastic/kibana/issues/95442